File: functions/trader_bridge.py
import json
import requests
import os
from pathlib import Path

def load_config():
    config_path = Path(__file__).parent.parent / "config" / "tradier_env.json"
    with open(config_path, 'r') as f:
        return json.load(f)

def get_tradier_client():
    config = load_config()
    current = config["environments"][config["active_env"]]
    
    session = requests.Session()
    session.headers.update({
        "Authorization": f"Bearer {current['token']}",
        "Accept": "application/json"
    })
    
    return session, current

def place_sniper_order(symbol, qty, order_type="market"):
    session, current = get_tradier_client()
    
    url = f"{current['base_url']}accounts/{current['brokerage_id']}/orders"
    
    payload = {
        "class": "equity",
        "symbol": symbol,
        "side": "buy",
        "quantity": qty,
        "type": order_type,
        "duration": "day"
    }
    
    try:
        response = session.post(url, data=payload)
        response.raise_for_status()
        data = response.json()
        
        if "order" in data and "id" in data["order"]:
            print(f"[Order Success] {qty}x {symbol} filled in {current['mode_label']}. Order ID: {data['order']['id']}")
            return data["order"]["id"]
        else:
            print(f"[Order Warning] Order sent but no ID returned for {symbol}")
            return None
            
    except Exception as e:
        error_detail = response.json() if 'response' in locals() else str(e)
        print(f"[Order Failed] Error in {current['mode_label']}: {error_detail}")
        raise e

if __name__ == "__main__":
    # Test Scout Order in Sandbox
    config = load_config()
    if config["active_env"] == "sandbox":
        print("Firing Test Scout Order in Python Sandbox...")
        try:
            place_sniper_order("/MNQ", 1)
        except:
            pass

File: trader_bridge.js
const fs = require('fs');
const axios = require('axios');
const path = require('path');

// Load environment from config
const configPath = path.join(__dirname, 'config', 'tradier_env.json');
const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
const current = config.environments[config.active_env];

const tradierClient = axios.create({
  baseURL: current.base_url,
  headers: {
    'Authorization': `Bearer ${current.token}`,
    'Accept': 'application/json'
  }
});

console.log(`[MCH] System Armed in ${current.mode_label} Mode.`);

/**
 * Executes the "Scout" or "Hammer" trade
 * @param {string} symbol - e.g., /MNQ
 * @param {number} qty - 1 for Scout, 2 for Hammer
 * @param {string} type - 'market' or 'limit'
 */
async function placeSniperOrder(symbol, qty, type = 'market') {
  try {
    const params = new URLSearchParams();
    params.append('class', 'equity');
    params.append('symbol', symbol);
    params.append('side', 'buy');
    params.append('quantity', qty);
    params.append('type', type);
    params.append('duration', 'day');

    const response = await tradierClient.post(`accounts/${current.brokerage_id}/orders`, params);
    
    if (response.data.order && response.data.order.id) {
        console.log(`[Order Success] ${qty}x ${symbol} filled in ${config.active_env}. Order ID: ${response.data.order.id}`);
        return response.data.order.id;
    } else {
        const errorMsg = response.data.errors ? JSON.stringify(response.data.errors) : "Unknown error";
        console.error(`[Order Failed] ${symbol} could not be filled: ${errorMsg}`);
        return null;
    }
  } catch (err) {
    const errorMsg = err.response ? JSON.stringify(err.response.data) : err.message;
    console.error(`[Order Failed] System Error in ${config.active_env}:`, errorMsg);
    throw err;
  }
}

module.exports = { place_sniper_order: placeSniperOrder, config: current };

// If run directly, fire a test Scout order if in Sandbox
if (require.main === module && config.active_env === 'sandbox') {
    console.log("Firing Test Scout Order in Sandbox...");
    placeSniperOrder('MNQ', 1).catch(e => {});
}
